docs/docs/Indexing_JSON.md

```diff
@@ -1,216 +1,367 @@
+linkTitle: Index JSON
+weight: 2
+description: How to index and search JSON documents
 ---
 
-In addition to indexing Redis hashes, RediSearch also indexes JSON. To index JSON, you must use the RedisJSON module.
+In addition to indexing Redis hashes, RediSearch can also index JSON documents. To index and search JSON, you need to install both the RediSearch and [RedisJSON](/docs/stack/json) modules.
 
 ## Prerequisites
 
-What do you need to start indexing JSON documents?
+Before you can index and search JSON documents, you need a database with either:
 
-- Redis 6.x or later
-- RediSearch 2.2 or later
-- RedisJSON 2.0 or later
+- [Redis Stack](/docs/stack), which automatically includes RediSearch and RedisJSON
 
-## How to index JSON documents
+- Redis v6.x or later and the following modules installed and enabled:
+   - RediSearch v2.2 or later
+   - RedisJSON v2.0 or later
 
-This section shows how to create an index.
+## Create index with JSON schema
 
-You can now specify `ON JSON` to inform RediSearch that you want to index JSON documents.
+When you create an index with the `FT.CREATE` command, include the `ON JSON` keyword to index any existing and future JSON documents stored in the database.
 
-For the `SCHEMA`, you can provide JSONPath expressions.
-The result of each `JSON Path` expression is indexed and associated with a logical name (`attribute`).
-This attribute (previously called `field`) is used in the query.
+To define the `SCHEMA`, you can provide [JSONPath](/docs/stack/json/path) expressions.
+The result of each JSONPath expression is indexed and associated with a logical name called an `attribute` (previously known as a `field`).
+You can use these attributes in queries.
 
-This is the basic syntax to index a JSON document:
+{{% alert title="Note" color="info" %}}
+Note: `attribute` is optional for `FT.CREATE`.
+{{% /alert %}}
 
-    FT.CREATE {index_name} ON JSON SCHEMA {json_path} AS {attribute} {type}
+Use the following syntax to create a JSON index:
 
-And here's a concrete example:
+```sql
+FT.CREATE {index_name} ON JSON SCHEMA {json_path} AS {attribute} {type}
+```
+
+For example, this command creates an index that indexes the name, description, and price of each JSON document that represents an inventory item:
+
+```sql
+127.0.0.1:6379> FT.CREATE itemIdx ON JSON PREFIX 1 item: SCHEMA $.name AS name TEXT $.description as description TEXT $.price AS price NUMERIC
+```
 
-    FT.CREATE userIdx ON JSON SCHEMA $.user.name AS name TEXT $.user.tag AS country TAG
+See [Index limitations](#index-limitations) for more details about JSON index `SCHEMA` restrictions.
 
-## Adding a JSON document to the index
+## Add JSON documents
 
-As soon as the index is created, any pre-existing JSON document, or any new JSON document added or modified, is
-automatically indexed.
+After you create an index, RediSearch automatically indexes any existing, modified, or newly created JSON documents stored in the database.
```

Comment:
@nermiller Should we note that for existing documents, indexing is done asynchronously in the background, and not necessarily ready immediately (`FT.INFO` has a field called `indexing` showing whether indexing is still ongoing or is it done)

Modified and newly created docs are indexed synchronously (index is ready by the time the add/modify command return)

(not related only to JSON docs)

Reply:
That would be helpful, @rrelledge.

Reply:
@oshadmi @nermiller Since this PR was already merged, I will open a new one to add this info.

Reply:
Here's the PR for this: https://github.com/RediSearch/RediSearch/pull/2949